File: app/src/main/java/com/botijasoftware/utils/animation/BoneHierarchy.java
package com.botijasoftware.utils.animation;

import java.util.ArrayList;

public class BoneHierarchy {

	public class BoneNode {

		public BoneNode parent;
		public Bone bone;
		public ArrayList<BoneNode> children = new ArrayList<BoneNode>();

		public BoneNode(Bone bone) {
			this.bone = bone;
		}

		public void addChilren(Bone bone) {

			BoneNode node = new BoneNode(bone);
			node.parent = this;
			children.add(node);
		}

	}

	public BoneNode root;

	public Bone findBone(String bonename) {

		return null;
	}
}

File: app/src/main/java/com/botijasoftware/utils/animation/BoneHierarchy.kt
package com.botijasoftware.utils.animation

import java.util.ArrayList

class BoneHierarchy {

    inner class BoneNode(var bone: Bone) {

        lateinit var parent: BoneNode
        var children = ArrayList<BoneNode>()

        fun addChilren(bone: Bone) {

            val node = BoneNode(bone)
            node.parent = this
            children.add(node)
        }

    }

    var root: BoneNode? = null

    fun findBone(bonename: String): Bone? {

        return null
    }
}
